ember-cli-build.js

```diff
@@ -1,26 +1,71 @@
+var Buffer = require('buffer').Buffer;
+var path = require('path');
 var Rollup = require('broccoli-rollup');
 var Funnel = require('broccoli-funnel');
 var MergeTrees = require('broccoli-merge-trees');
-var FileCreator = require('broccoli-file-creator');
-var replace = require('rollup-plugin-replace');
+var replace = require('broccoli-string-replace');
+var typescript = require('broccoli-typescript-compiler').typescript;
+var buble = require('rollup-plugin-buble');
+var fs = require('fs');
+
+var SOURCE_MAPPING_DATA_URL = '//# sourceMap';
+SOURCE_MAPPING_DATA_URL += 'pingURL=data:application/json;base64,';
 
 module.exports = function () {
+  var src = new MergeTrees([
+    new Funnel(path.dirname(require.resolve('@types/qunit/package')), {
+      include: [ "**/*.d.ts" ],
+      destDir: "qunit",
+      annotation: "qunit typescript typings"
+    }),
+    replace(new Funnel("lib", {
+      include: [ "**/*.ts" ]
+    }), {
+      files: ["route-recognizer.ts"],
+      pattern: {
+        match: /VERSION_STRING_PLACEHOLDER/g,
+        replacement: require("./package").version
+      }
+    }),
+    new Funnel("tests", {
+      include: [ "**/*.ts" ],
+      destDir: "tests"
+    })
+  ]);
+
+  var compiled = typescript(src, {
+    annotation: 'compile route-recognizer.ts',
+    tsconfig: {
+      compilerOptions: {
+        module: "es2015",
+        moduleResolution: "node",
+        target: "es2015",
+        declaration: true,
+        strictNullChecks: true,
+        inlineSourceMap: true,
+        inlineSources: true,
+        baseUrl: ".",
+        paths: {
+          "route-recognizer": ["route-recognizer.ts"]
+        }
+      },
+      files: [
+        "route-recognizer.ts",
+        "qunit/index.d.ts",
+        "tests/index.ts"
+      ]
+    }
+  });
   return new MergeTrees([
-    new Rollup('lib', {
+    new Rollup(compiled, {
       annotation: 'route-recognizer.js',
       rollup: {
         entry: 'route-recognizer.js',
-        plugins: [replace({
-          VERSION_STRING_PLACEHOLDER: require('./package.json').version
-        })],
+        plugins: [ loadWithInlineMap(), buble() ],
```

Comment:
is there a reason we can't use broccoli-babel-transpiler here? We have seen issues with buble doing funky things in heimdall, and also i don't believe it benefits from a persistent cache.

Absolutely not a big deal, just curious. (no need to change if you dont feel there is an issue)